runtime/ftplugin/java.vim

```diff
@@ -78,6 +78,10 @@ setlocal comments& comments^=:///,sO:*\ -,mO:*\ \ ,exO:*/
 
 setlocal commentstring=//\ %s
 
+if expand('%:e') ==# 'jsh'
+  match Comment /^#!.*/
```

Comment:
Please can you limit the match to some arbitrary top lines,  
e.g. `\%<32l^#!.*`, and make the indentation conform to the  
modeline?